pkg/frontend/middleware/log.go

```diff
@@ -52,21 +52,27 @@ func Log(baseLog *logrus.Entry) func(http.Handler) http.Handler {
 			r.Body = &logReadCloser{ReadCloser: r.Body}
 			w = &logResponseWriter{ResponseWriter: w, statusCode: http.StatusOK}
 
+			// correlationID is set by arm frontend. Local RP will not have this set
 			correlationID := r.Header.Get("X-Ms-Correlation-Request-Id")
+			if correlationID == "" {
+				correlationID = uuid.NewV4().String()
+			}
 
 			requestID := uuid.NewV4().String()
 			w.Header().Set("X-Ms-Request-Id", requestID)
 
 			fields := logrus.Fields{
-				"correlation_id": correlationID,
-				"request_id":     requestID,
+				"correlation_id":    correlationID,
+				"request_id":        requestID,
+				"client_request_id": r.Header.Get("X-Ms-Client-Request-Id"),
```

Comment:
no need to log this

Reply:
`If the caller provides this header – the resource provider *must* log this with their traces to facilitate tracing a single request` Is this not applicable for us? 
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#client-request-headers

Reply:
sure, we can do that one as well.  3 then.